### DISCUSSION TITLE
Form-level error

### DISCUSSION THREAD

**Discussion by fakkio** [react-hook-form/react-hook-form#3190] [Q&A]
Hi,
is it possible to set an error at form-level instead of input-level?

Thanks.

**Comment by bluebill1049** [comment]
`setError`

**Reply by fakkio** [reply]
Yeah, i kwon the `setError` method, but it sets a field error, i'd like to set a form error, i.e. an error that isn't referred to a particular field.

**Reply by bluebill1049** [reply]
if you need form level error, i would recommend to use `resolver`, you will be able to validate the entire form at your own control.

**Reply by fakkio** [reply]
I am using it. But then i can assign error only to a specific filed due to the `errors` type that is:
```
FieldErrors<FormData>
```

**Reply by bluebill1049** [reply]
> But then i can assign error only to a specific filed due to the `errors` type that is:

what do you mean?

**Reply by robinmetral** [reply]
I think that what they mean is that errors can only be assigned to a specific field.

So if the fields are:

```ts
type FormFields = {
  firstName: string;
  lastName: string;
}
```

we can assign errors to either field with for example:

```ts
setError("firstName", { type: "validate", message: "Invalid first name" })
```

...but we can't assign an error to the entire form, because setError expects a field name.

The use case here could be that we want to show a global "Internal Server Error" if we get a 500 from async validation. This would not be linked to any specific field, but to the global form submission.

Is there a mechanism to handle this?

**Reply by bluebill1049** [reply]
Is this article helps? https://www.carlrippon.com/react-hook-form-server-validation/

**Reply by robinmetral** [reply]
I've seen it before because it was linked in the docs, but it looks to me that it's simply mapping server-side errors to the RHF field-level errors, so that wouldn't solve this.

For now I'm implementing it manually in my app, something like:

```ts
const [submissionError, setSubmissionError] = useState<string>(null)
```

And if there's a general 500 from the server, I store it there and render it in the app accordingly.

A 500 can't be mapped to a single field

---

Edit: yep, looks like the post is just mapping to field-level errors:
> we can iterate through the server validation errors, calling the setError method, mapping to the React Hook Form’s error format.

**Reply by bluebill1049** [reply]
maybe we should catch any error which gets thrown in the handlSubmit, and the error will be `errors.serverError`. It's just an idea.

**Reply by robinmetral** [reply]
This could work :slightly_smiling_face:

**Reply by bluebill1049** [reply]
here is the feature request: https://github.com/react-hook-form/react-hook-form/projects/1 will have an RFC after v7 is released.

**Reply by fakkio** [reply]
> > But then i can assign error only to a specific filed due to the `errors` type that is:
> 
> what do you mean?

Hey, sorry for the huuuge late.

my use case was that I had a form where users choose how to divide a quantity. Let's say you have 5 input that have to sum up to 100 %, if they don't meet 100, where can i put the error object? It's not an error specific to some field.
When I used redux-form there was a `_error` that could contain an arbitrary error and then I could visualize the error where I wanted to (usually just under the form buttons)

I hope I made myself clear this time.

Fabio.

**Reply by bluebill1049** [reply]
If you need to have it on the form level, I think it's easier to build a schema and use `trigger` to return a group/form level errors.

**Reply by fakkio** [reply]
> If you need to have it on the form level, I think it's easier to build a schema and use `trigger` to return a group/form level errors.

got it, Thank you 👍
